code/game/machinery/computer/atmos_control.dm

```diff
@@ -0,0 +1,589 @@
+			"carbon_dioxide" = airalarm_presets[preset].carbon_dioxide,
+			"plasma" = airalarm_presets[preset].plasma,
+			"n2o" = airalarm_presets[preset].n2o,
+			"other" = airalarm_presets[preset].other,
+			"pressure" = airalarm_presets[preset].pressure,
+			"temperature" = airalarm_presets[preset].temperature,
+			"target_temperature" = airalarm_presets[preset].target_temperature,
+			"scrubbers_gases" = airalarm_presets[preset].scrubbers_gases
+		)
+	data["presets"] = tmplist
+
+	if(selected_preset)
+		data["selected_preset"] = list(
+			"name" = selected_preset.name,
+			"desc" = selected_preset.desc,
+			"core" = selected_preset.core,
+			"oxygen" = selected_preset.oxygen,
+			"nitrogen" = selected_preset.nitrogen,
+			"carbon_dioxide" = selected_preset.carbon_dioxide,
+			"plasma" = selected_preset.plasma,
+			"n2o" = selected_preset.n2o,
+			"other" = selected_preset.other,
+			"pressure" = selected_preset.pressure,
+			"temperature" = selected_preset.temperature,
+			"target_temperature" = selected_preset.target_temperature,
+			"scrubbers_gases" = selected_preset.scrubbers_gases
+		)
+		data["selected_preset_name"] = selected_preset.name
+	else
+		data["selected_preset"] = null
+		data["selected_preset_name"] = null
+	
+	if(log_in_id && (access_ce in log_in_id.GetAccess()) || emagged || user.hasFullAccess())
+		data["admin_access"] = TRUE
+	else
+		data["admin_access"] = FALSE
+		screen = ACA_SCREEN_DETAILSVIEW //this dumb hack stops unauthorized cards from seeing shit they shouldn't
+
+	data["aca_screen"] = screen //aca_screen so we don't conflict with air alarms, which already use screen
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "atmos_control.tmpl", name, 900, 800)
+		// adding a template with the key "mapContent" enables the map ui functionality
+		ui.add_template("mapContent", "atmos_control_map_content.tmpl")
+		// adding a template with the key "mapHeader" replaces the map header content
+		ui.add_template("mapHeader", "atmos_control_map_header.tmpl")
+		ui.set_show_map(FALSE)
+		ui.set_initial_data(data)
+		ui.open()
+	ui.set_auto_update(!!current)
+
+//a bunch of this is copied from atmos alarms
+/obj/machinery/computer/atmoscontrol/Topic(href, href_list)
+	if(..())
+		return 1
+
+	if(href_list["set_screen"])
+		screen = href_list["set_screen"]
+		return 1
+
+	if(href_list["login"])
+		if(log_in_id || emagged)
+			return 1
+		var/mob/M = usr
+		var/obj/item/weapon/card/id/I = M.get_active_hand()
+		if (istype(I, /obj/item/device/pda))
+			var/obj/item/device/pda/pda = I
+			I = pda.id
+		if (istype(I,/obj/item/weapon/card/emag))
+			emag(I, usr)
+		if (I && istype(I))
+			log_in_id = I
+		return 1
+
+	if(href_list["logout"])
+		if(!emagged)
+			log_in_id = null
+		return 1
+
+	if(href_list["close"])
+		if(usr.machine == src)
+			usr.unset_machine()
+		return 1
+
+	if(href_list["reset"])
+		current = null
+		return TRUE
```

Comment:
weird inconsistency that you swap between 1 and TRUE

Reply:
The ones that use TRUE are legacy code while the ones that use 1 were written by me.

Reply:
that's fine just pick one direction and consistify them

Reply:
I went with 1 because that's what nanouis in general use.